### PR TITLE
pkg/esp32_sdk: Update version to 4.4.1 

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This updates the ESP-IDF package to the latest tagged version.
Only the [esp32-wifi-lib](https://github.com/espressif/esp32-wifi-lib) could not be updated to the latest version as the current RIOT drivers will crash after [c73ee33](https://github.com/espressif/esp32-wifi-lib/commit/c73ee337c334597378e79219c9f59525116437d2) - `Remove legacy system event framework(1948a1c7)`.

### Testing procedure

`examples/gnrc_border_router` still works with `esp_now` and `esp_wifi`.

```
2022-06-02 23:55:01,101 # main(): This is RIOT! (Version: 2022.07-devel-646-g95f8ce-pkg/esp32_sdk-4.4.1)
2022-06-02 23:55:01,105 # RIOT border router example application
2022-06-02 23:55:01,107 # All up, running the shell now

2022-06-02 23:57:14,345 # Iface  10  HWaddr: 3C:71:BF:9E:13:FC  Channel: 1  Link: up 
2022-06-02 23:57:14,349 #           L2-PDU:1500  MTU:1492  HL:255  RTR  
2022-06-02 23:57:14,352 #           Source address length: 6
2022-06-02 23:57:14,354 #           Link type: wireless
2022-06-02 23:57:14,360 #           inet6 addr: fe80::3e71:bfff:fe9e:13fc  scope: link  VAL
2022-06-02 23:57:14,367 #           inet6 addr: 2001:16b8:45eb:4800:3e71:bfff:fe9e:13fc  scope: global  VAL
2022-06-02 23:57:14,370 #           inet6 group: ff02::2
2022-06-02 23:57:14,373 #           inet6 group: ff02::1
2022-06-02 23:57:14,376 #           inet6 group: ff02::1:ff9e:13fc
2022-06-02 23:57:14,377 #           
2022-06-02 23:57:14,381 # Iface  9  HWaddr: 3E:71:BF:9E:13:FC  Channel: 1 
2022-06-02 23:57:14,385 #           L2-PDU:249  MTU:1280  HL:64  RTR  
2022-06-02 23:57:14,390 #           RTR_ADV  6LO  Source address length: 6
2022-06-02 23:57:14,392 #           Link type: wireless
2022-06-02 23:57:14,398 #           inet6 addr: fe80::3c71:bfff:fe9e:13fc  scope: link  VAL
2022-06-02 23:57:14,405 #           inet6 addr: 2001:16b8:45eb:48fc:3c71:bfff:fe9e:13fc  scope: global  VAL
2022-06-02 23:57:14,408 #           inet6 group: ff02::2
2022-06-02 23:57:14,410 #           inet6 group: ff02::1
2022-06-02 23:57:14,414 #           inet6 group: ff02::1:ff9e:13fc
2022-06-02 23:57:14,415 #           
> ping 2600::
2022-06-02 23:57:18,178 # ping 2600::
2022-06-02 23:57:18,401 # 12 bytes from 2600::: icmp_seq=0 ttl=53 time=217.642 ms
2022-06-02 23:57:19,418 # 12 bytes from 2600::: icmp_seq=1 ttl=53 time=234.909 ms
2022-06-02 23:57:20,341 # 12 bytes from 2600::: icmp_seq=2 ttl=53 time=158.359 ms
2022-06-02 23:57:20,342 # 
2022-06-02 23:57:20,344 # --- 2600:: PING statistics ---
2022-06-02 23:57:20,349 # 3 packets transmitted, 3 packets received, 0% packet loss
2022-06-02 23:57:20,354 # round-trip min/avg/max = 158.359/203.636/234.909 ms

2022-06-03 00:07:34,851 # ping ff02::1%9
2022-06-03 00:07:34,864 # 12 bytes from fe80::807d:3aff:fe7f:e01d%9: icmp_seq=0 ttl=64 time=7.463 ms
2022-06-03 00:07:35,866 # 12 bytes from fe80::807d:3aff:fe7f:e01d%9: icmp_seq=1 ttl=64 time=9.740 ms
2022-06-03 00:07:36,863 # 12 bytes from fe80::807d:3aff:fe7f:e01d%9: icmp_seq=2 ttl=64 time=6.814 ms
2022-06-03 00:07:36,863 # 
2022-06-03 00:07:36,866 # --- ff02::1%9 PING statistics ---
2022-06-03 00:07:36,871 # 3 packets transmitted, 3 packets received, 0% packet loss
2022-06-03 00:07:36,875 # round-trip min/avg/max = 6.814/8.005/9.740 ms


```

### CODE REVIEW

**Comment by gschorcht** [comment]
Please don't try to upgrade the ESP-IDF to a newer version in this stage of migration process because it is in contradiction to the subsequent PRs #17841, #17842 and #17844

**Comment by gschorcht** [comment]
Please don't try to upgrade the ESP-IDF to a newer version in this stage of migration process because it is in contradiction to the subsequent PRs https://github.com/RIOT-OS/RIOT/pull/17841, https://github.com/RIOT-OS/RIOT/pull/17842 and https://github.com/RIOT-OS/RIOT/pull/17844. Even though it seems to work with current master, there are some changes that require again expensive work with these subsequent PRs.

**Comment by gschorcht** [comment]
@benpicco What was the motivation to suggest this ESP-IDF upgrade?

**Comment by gschorcht** [comment]
I personally would prefer to merge all subsequent PR including the ESP32-S3 support first and to try to upgrade ESP-IDF later directly to version v5.0 which includes the support for the RISC-V based ESP32-H2 variant.

**Comment by benpicco** [comment]
I just saw that there was a bugfix release and thought bugfixes are always good to have in.
I also wanted to see how easy it is to upgrade to a new version of the SDK.

File: pkg/esp32_sdk_lib_wifi/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PKG_NAME=esp32_sdk_lib_wifi
 PKG_URL=https://github.com/espressif/esp32-wifi-lib
-# This is a version in the v4.4 release branch
-PKG_VERSION=cd7d14917f2c3d0ea4382f4a188cb290304faf47
+# This is the last version with the legacy system event framework
+PKG_VERSION=81768e63548385be79e7b35828832a53faba4393
```
Comment (gschorcht) [inline pkg/esp32_sdk_lib_wifi/Makefile:4]: This commit and commit [c73ee33](https://github.com/espressif/esp32-wifi-lib/commit/c73ee337c334597378e79219c9f59525116437d2) - `Remove legacy system event framework(1948a1c7) ` are commits from the master branch that do not match the ESP-IDF 4.4 branch but are for the upcoming ESP-IDF version 5.0. The latest matching commit is [25e90ea](https://github.com/espressif/esp32-wifi-lib/commit/25e90eac783e0aeb6772d91585c671e00388126d) ([release/v4.4](25e90eac783e0aeb6772d91585c671e00388126d)).

BTW, with the ESP32-S3 port, this library is upgraded to commit [1b0fcf9](https://github.com/espressif/esp32-wifi-lib/commit/1b0fcf9eeb96ddc4d72ab5850a0bf9e95e289da4).

**Comment by gschorcht** [comment]
Ok, I could at least compile and test `tests/shell` as well as compile `examples/gnrc_networking` with PR #17841 on top of this PR.

**Comment by gschorcht** [comment]
I could also compile and test `tests/shell` and compile `examples/gnrc_networking` with PR #17842 and #17844 (ESP32-C3) on top of this PR.

**Comment by gschorcht** [comment]
@benpicco The ESP32-S3 port also seems to still work (without detailed testing). So I have to revise my first reaction. It seems that we can give this upgrade a try. Please upgrade the packages to the right versions.

**Comment by gschorcht** [comment]
To fix the [problem](https://ci.riot-os.org/RIOT-OS/RIOT/18165/cb0d56998bbd532332f001b5c38ff245f63f2a29/output/compile/examples/gnrc_minimal/esp32-ci:gnu.txt), we have to define these symbols like the others in `cpu/esp32/esp-idf/esp_idf_support.c`:
```diff
diff --git a/cpu/esp32/esp-idf/esp_idf_support.c b/cpu/esp32/esp-idf/esp_idf_support.c
index 9a0083216a..68df412207 100644
--- a/cpu/esp32/esp-idf/esp_idf_support.c
+++ b/cpu/esp32/esp-idf/esp_idf_support.c
@@ -64,6 +64,11 @@ int g_wpa_ca_cert_len;
 char *g_wpa_ttls_phase2_type;
 bool g_wpa_suiteb_certification;
 
+char *g_wpa_phase1_options;
+
+uint8_t *g_wpa_pac_file;
+int g_wpa_pac_file_len;
+
 /*
  * provided by: /path/to/esp-idf/components/log/log_freertos.c
  */
```

Fortunately we have added the tests compilation to the board `esp32-ci` with PR #18158. Otherwise we would not have noticed it.

**Review by gschorcht — APPROVED** [review]

**Pull request merged**